src/main/java/duke/parsers/FileParser.java

```diff
@@ -0,0 +1,62 @@
+package duke.parsers;
+
+import duke.tasks.Event;
+import duke.tasks.ToDo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.time.LocalDate;
+import java.util.Scanner;
+import duke.tasks.Deadline;
+import duke.tasks.TaskList;
+
+/**
+ * Class to parse text file and generates a TaskList for modifications from the user.
+ */
+public class FileParser {
+
+    public File file;
+    public TaskList taskList;
+
+    public FileParser(File file) {
+        this.file = file;
+        this.taskList = new TaskList();
+    }
+
+    /**
+     * Parses file object
+     * @param file File object to be passed
+     * @throws FileNotFoundException if file object is not found.
+     */
+    public void parseFile(File file) throws FileNotFoundException {
```

Comment:
Very good that you specify what the method is parsing. Clear naming, no common mistakes 👍